packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -604,12 +604,6 @@ function getErrorBoundaryVM(vm: VM): VM | undefined {
 // NOTE: we should probably more this routine to the synthetic shadow folder
 // and get the allocation to be cached by in the elm instead of in the VM
 export function allocateInSlot(vm: VM, children: VNodes) {
-    if (process.env.NODE_ENV !== 'production') {
```

Comment:
I was looking through the code, and this assertion doesn't make sense anymore as the `vm.cmpSlots` is always initialized.